### PR TITLE
fix: Fix type upgrade issues

### PR DESCRIPTION
**Pull request by alanbsmith**
## Summary
This PR fixes:

* **Banner** - remove hardcoded font size + weight & use correct type level
* **Breadcrumbs** - fix Link styles
* **CookieBanner** - fix setting button link styles
* **Deprecated Button** - hardcode values to prevent changes
* **Pagination** - fix Additional Details and Go To Label font sizes
* **Tab** - fix font-weight (500 instead of 700)
* **Toast** - fix Action button link styles
* **TooltipContainer** - remove hardcoded 13px value

Also includes the following from #1096 

- [x] Corrects some of the manual migration done to the tokens stories in #1078 
- [x] Fixes missing logic in type codemod that was preventing the font weight from being preserved for h3 and h4 headings that were specifying the type object in a `styled()` argument
- [x] Run codemod update (or manually update the issues)
- [x] BREAKING: Removes unintended `0.01rem` letter spacing from `body.m` and `body.l`
- [x] Upsize toast text to 14px because 12px was too small

And updates the v5 migration guide - more info for link variant migration

## Checklist

- [ ] branch has been rebased on the latest master commit
- [ ] tests are changed or added
- [ ] `yarn test` passes
- [ ] all (dev)dependencies that the module needs is added to its `package.json`
- [ ] code has been documented and, if applicable, usage described in README.md
- [ ] module has been added to `canvas-kit-react` and/or `canvas-kit-css` universal modules, if
      applicable
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](../API_PATTERN_GUIDELINES.md)

### CODE REVIEW

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx [hunk 1]
```diff
@@ -373,25 +373,37 @@ const monoStyles = {fontFamily: type.properties.fontFamilies.monospace};
...
 
-const Link = styled('a')(
-  type.variant.link,
-);
+const Link = styled('a')(type.variant.link);
 
-return (
-  <Link href="https://workday.github.io/canvs-kit">View docs</Link>
-);
+return <Link href="https://workday.github.io/canvs-kit">View docs</Link>;
```
Comment (willklein) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:384]:
```suggestion
return <Link href="https://workday.github.io/canvas-kit">View docs</Link>;
```

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx [hunk 2]
```diff
@@ -373,25 +373,37 @@ const monoStyles = {fontFamily: type.properties.fontFamilies.monospace};
...
+
+// v5
+import {Hyperlink} from '@workday/canvas-kit-labs-react/button';
+
+return <Hyperlink href="https://workday.github.io/canvs-kit">View docs</Hyperlink>;
```
Comment (willklein) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:389]:
```suggestion
return <Hyperlink href="https://workday.github.io/canvas-kit">View docs</Hyperlink>;
```

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx [hunk 3]
```diff
@@ -373,25 +373,37 @@ const monoStyles = {fontFamily: type.properties.fontFamilies.monospace};
...
+import {type} from '@workday/canvas-kit-labs-react/tokens';
 import {Hyperlink} from '@workday/canvas-kit-labs-react/button';
 
-return (
-  <Hyperlink href="https://workday.github.io/canvs-kit">View docs</Link>
+// Remove `color` from type styles to prevent the color from overriding the link color
+const {color, ...headingLargeStyles} = type.levels.heading.large;
+
+const HeadingLink = () => (
+  <Hyperlink css={headingLargeStyles} href="https://workday.github.io/canvs-kit">
```
Comment (willklein) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:404]:
```suggestion
  <Hyperlink css={headingLargeStyles} href="https://workday.github.io/canvas-kit">
```

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx [hunk 4]
```diff
@@ -1463,4 +1475,7 @@ const MySkeleton = () => (
 
 Additionally, the skeleton animation has been updated from a diagonal sheen, or shimmer, to fading
 the opacity of the entire shape(s) in and out.
+
+```
```
Comment (willklein) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:1479]: This keeps happening. Not sure what's up.

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx (hunk 2 above)
Comment (alanbsmith) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx]:
```suggestion
return <Hyperlink href="https://workday.github.io/canvas-kit">View docs</Hyperlink>;
```

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx (hunk 1 above)
Comment (alanbsmith) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:384]:
```suggestion
return <Link href="https://workday.github.io/canvas-kit">View docs</Link>;
```

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx (hunk 4 above)
Comment (NicholasBoll) [reply modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:1479]: I think a formatter is closing the doc block

File: modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx [hunk 5]
```diff
@@ -1463,4 +1475,7 @@ const MySkeleton = () => (
...
+
 ```
```
Comment (NicholasBoll) [inline modules/docs/mdx/5.0-MIGRATION-GUIDE.mdx:1481]:
```suggestion

```

**Review by anicholls — APPROVED** [review]

**Pull request merged**